code/modules/mob/living/simple_animal/friendly/butterfly.dm

```diff
@@ -32,3 +32,124 @@
 
 /mob/living/simple_animal/butterfly/bee_friendly()
 	return TRUE //treaty signed at the Beeneeva convention
+
+/mob/living/simple_animal/hostile/fly
+	name = "fly"
+	desc = "We went to space to escape flies, but they always seem to find us."
+	icon_state = "fly"
+	icon_living = "fly"
+	icon_dead = "fly_dead"
+	icon = 'icons/mob/bees.dmi'
+	gender = NEUTER
+	speak_emote = list("buzzes")
+	emote_hear = list("buzzes")
+	turns_per_move = 0
+	melee_damage_lower = 0
+	melee_damage_upper = 0
+	retreat_distance = 2
+	minimum_distance = 2
+	friendly = "annoys"
+	attacktext = "seriously annoys"
+	response_help  = "shoos"
+	response_disarm = "swats away"
+	response_harm   = "squashes"
+	maxHealth = 1
+	health = 1
+	move_to_delay = 0
+	obj_damage = 0
+	environment_smash = ENVIRONMENT_SMASH_NONE
+	mouse_opacity = MOUSE_OPACITY_OPAQUE
+	pass_flags = PASSTABLE | PASSGRILLE | PASSMOB
+	density = FALSE
+	mob_size = MOB_SIZE_TINY
+	mob_biotypes = list(MOB_ORGANIC, MOB_BUG)
+	movement_type = FLYING
+	search_objects = 1 //have to find those plant trays!
+
+/mob/living/simple_animal/hostile/fly/Initialize()
+	. = ..()
+	AddComponent(/datum/component/swarming)
+
+/mob/living/simple_animal/hostile/fly/time
+	name = "time fly"
+	desc = "it's just a jump to the left..."
+	attacktext = "zaps"
+	icon_state = "timefly"
+	icon_living = "timefly"
+	icon_dead = "timefly_dead"
+	var/obj/effect/proc_holder/spell/targeted/timewarp/twarp
+	var/datum/action/innate/timewarp/twarpself
+
+/mob/living/simple_animal/hostile/fly/time/Initialize()
+	. = ..()
+	twarp = new
+	twarpself = new
+	twarpself.Grant(src)
+
+/mob/living/simple_animal/hostile/fly/time/AttackingTarget()
+	if(. && isliving(target))
+		var/mob/living/L = target
+		if(twarp)
+			twarp.cast(L, src)
+
+/datum/action/innate/timewarp/Activate()
+	if(!istype(owner, /mob/living/simple_animal/hostile/fly/time))
+		return
+	var/mob/living/simple_animal/hostile/fly/time/TF = owner
+	if(TF.twarp)
+		TF.twarp.cast(TF, TF)
+
+/obj/effect/proc_holder/spell/targeted/timewarp //this spell exists (probably in nullspace) but is not granted to any character, it is triggered by the time fly's actions. good god old armhulen you were SO ahead of your time
+	name = "Time Warp!"
+	desc = "I REMEMBERRRRRR DOIN THE TIIIME WARRRRRP DRINKING THOSE MOMENTS WHEEEEEEEEEN THE BLACKNESS WOULD HIT MEEEEEE AND THE VOID WOULD BE CAAAAAAAALLING"
```

Comment:
While excellent, maybe shorten it a bit...?

Maybe `It's just a jump to the left.`

Reply:
On initialize have it pick a line from the song.

Reply:
I can only get away with this because the spell itself sits in nullspace. Seeing the desc is impossible

Reply:
you're telling me the action button doesn't inherit this?